### ISSUE TITLE
All examples dump core

### ISSUE THREAD

**Issue by jn0** [irrtoolset/irrtoolset#61]
Fresh clone, just built (forced LANG to not clobber output with non-latin chars):

```
jno@jno-ThinkPad-L13:~/src/irrtoolset/examples[master:3]$ env LANG=C bash -x ./build_examples.sh 
+ echo building IOS example config into: ios-example-output.txt
building IOS example config into: ios-example-output.txt
+ ../src/rtconfig/rtconfig -h whois.radb.net -f private_irr_objects.txt -p 43 -protocol rawhoisd -config cisco -supress_martian -cisco_peer_templates -cisco_use_prefix_lists -cisco_empty_lists
./build_examples.sh: line 4: 30498 Segmentation fault      (core dumped) ../src/rtconfig/rtconfig -h whois.radb.net -f private_irr_objects.txt -p 43 -protocol rawhoisd -config cisco -supress_martian -cisco_peer_templates -cisco_use_prefix_lists -cisco_empty_lists < ios-example-input.cfg > ios-example-output.txt
+ echo building IOS XR example config into: xr-example-output.txt
building IOS XR example config into: xr-example-output.txt
+ ../src/rtconfig/rtconfig -h whois.radb.net -f private_irr_objects.txt -p 43 -protocol rawhoisd -config ciscoxr -supress_martian
./build_examples.sh: line 8: 30500 Segmentation fault      (core dumped) ../src/rtconfig/rtconfig -h whois.radb.net -f private_irr_objects.txt -p 43 -protocol rawhoisd -config ciscoxr -supress_martian < xr-example-input.cfg > xr-example-output.txt
+ echo building JunOS example config into: junos-example-output.txt
building JunOS example config into: junos-example-output.txt
+ ../src/rtconfig/rtconfig -h whois.radb.net -f private_irr_objects.txt -p 43 -protocol rawhoisd -config junos -junos_replace -junos_and_not_or -supress_martian
./build_examples.sh: line 12: 30504 Segmentation fault      (core dumped) ../src/rtconfig/rtconfig -h whois.radb.net -f private_irr_objects.txt -p 43 -protocol rawhoisd -config junos -junos_replace -junos_and_not_or -supress_martian < junos-example-input.cfg > junos-example-output.txt
jno@jno-ThinkPad-L13:~/src/irrtoolset/examples[master:3]$ LANG=C ls -ltr
total 24
-rw-rw-r-- 1 jno jno 1601 May 24 11:30 xr-example-input.cfg
-rw-rw-r-- 1 jno jno 1549 May 24 11:30 private_irr_objects.txt
-rw-rw-r-- 1 jno jno 3233 May 24 11:30 junos-example-input.cfg
-rw-rw-r-- 1 jno jno 1720 May 24 11:30 ios-example-input.cfg
-rwxrwxr-x 1 jno jno  961 May 24 11:30 build_examples.sh
-rw-rw-r-- 1 jno jno  647 May 24 11:30 README
-rw-rw-r-- 1 jno jno    0 May 24 11:41 ios-example-output.txt
-rw-rw-r-- 1 jno jno    0 May 24 11:41 xr-example-output.txt
-rw-rw-r-- 1 jno jno    0 May 24 11:41 junos-example-output.txt
jno@jno-ThinkPad-L13:~/src/irrtoolset/examples[master:3]$ 
```

**Comment by nickhilliard** [comment]
what platform / gcc version?

**Comment by nickhilliard** [comment]
@jn0 - what linux distribution and version did you see this problem on?

**Comment by nickhilliard** [comment]
timeout, closing.  I'll re-open it if you can provide feedback.

**Comment by jn0** [comment]
Ooops. Missed reply :(

```
jno@jno-ThinkPad-L13:~$ lsb_release -a; uname -a; gcc --version
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.2 LTS
Release:	20.04
Codename:	focal
Linux jno-ThinkPad-L13 5.8.0-59-generic #66~20.04.1-Ubuntu SMP Thu Jun 17 11:14:10 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
gcc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0
Copyright (C) 2019 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by afpd** [comment]
Probably 'that optimization' issue again? https://github.com/irrtoolset/irrtoolset/issues/56 and similar.

**Comment by nickhilliard** [comment]
> Probably 'that optimization' issue again? #56 and similar.

yep, that's the one.  fixed in 6e0aa5a8486bbc2248f3945bf8e942d96331ed60. @jn0 thx for logging this.
